py_v_sdk/contract/atomic_swap_ctrt.py

```diff
@@ -151,14 +243,331 @@ async def token_id(self) -> str:
-        get_tok_bal queries & returns the balance of the token deposited into the contract.
+        get_swap_balance queries & returns the balance of the token deposited into the contract.
 
         Args:
             addr (str): The account address that deposits the token.
 
         Returns:
-            int: The balance of the token.
+            md.Token: The balance of the token.
+        """
+        bal = await self._query_db_key(self.DBKey.for_token_balance(addr))
+
+        resp = await self.chain.api.ctrt.get_tok_info(await self.token_id)
+        unit = resp["unity"]
+        return md.Token.for_amount(bal, unit)
+
+    async def get_swap_owner(self, tx_id: str) -> md.Addr:
+        """
+        get_swap_owner queries & returns the address of swap owner.
+
+        Args:
+            tx_id (str): The lock transaction id.
+
+        Returns:
+            md.Addr: The address of the swap owner.
+        """
+        owner_addr = await self._query_db_key(self.DBKey.for_swap_owner(tx_id))
+
+        return md.Addr(owner_addr)
+
+    async def get_swap_recipient(self, tx_id: str) -> md.Addr:
+        """
+        get_swap_recipient queries & returns the address of swap recipient.
+
+        Args:
+            tx_id (str): The lock transaction id.
+
+        Returns:
+            md.Addr: The address of the recipient.
+        """
+        recipient_addr = await self._query_db_key(self.DBKey.for_swap_recipient(tx_id))
+
+        return md.Addr(recipient_addr)
+
+    async def get_swap_puzzle(self, tx_id: str) -> str:
+        """
+        get_swap_puzzle queries & returns the hashed secret.
+
+        Args:
+            tx_id (str): The lock transaction id.
+
+        Returns:
+            str: The puzzle.
+        """
+        puzzle = await self._query_db_key(self.DBKey.for_puzzle(tx_id))
+
+        return puzzle
+
+    async def get_swap_amount(self, tx_id: str) -> md.Token:
+        """
+        get_swap_amount queries & returns the balance that the token locked.
+
+        Args:
+            tx_id (str): The lock transaction id.
+
+        Returns:
+            int: The balance of the token locked.
         """
-        return await self._query_db_key(self.DBKey.for_token_balance(addr))
+        amount = await self._query_db_key(self.DBKey.for_swap_amount(tx_id))
+
+        resp = await self.chain.api.ctrt.get_tok_info(await self.token_id)
+        unit = resp["unity"]
+        return md.Token.for_amount(amount, unit)
+
+    async def get_swap_expired_time(self, tx_id: str) -> md.VSYSTimestamp:
+        """
+        get_swap_expired_time queries & returns the expired timestamp.
+
+        Args:
+            tx_id (str): The lock transaction id.
+
+        Returns:
+            md.VSYSTimestamp: The expired timestamp.
+        """
+        expired_time = await self._query_db_key(self.DBKey.for_swap_expired_time(tx_id))
+
+        return md.VSYSTimestamp(int(expired_time))
+
+    async def get_swap_status(self, tx_id: str) -> str:
```

Comment:
The type hint here (`str`) doesn't match the one in the docstring (`bool`). If the `status` is like `'true'` and `'false'`, can return like `return status == 'true'` so that the return type is `bool`.